### ISSUE TITLE
Error in demo

### ISSUE THREAD

**Issue by MVelascoFlores** [fleaflet/flutter_map#245]
I copy and paste the demo (editing the token) but the map is blank, i can see the flutter logo 
the console print this "Another exception was thrown: Couldn't download or retreive file."

**Comment by eliburgi** [comment]
I get the same exception :(

**Comment by madeinquant** [comment]
The map is black and I have a make a simular case with mapbox access token.

**Comment by madeinquant** [comment]
> I copy and paste the demo (editing the token) but the map is blank, i can see the flutter logo
> the console print this "Another exception was thrown: Couldn't download or retreive file."

I find a solution in this video.
https://www.youtube.com/watch?v=oc1mktx0Zn8

**Comment by johnpryan** [comment]
I can't reproduce this. can you provide a small example that reproduces the issue?

**Comment by mulderpf** [comment]
What is the solution in the video?  Or at least where in the video can I find it?

I have seen the same issue in production and found the same symptoms on Stackoverflow: https://stackoverflow.com/questions/56098042/any-clue-for-crash-report-string-couldnt-download-or-retreive-file-got-fro/57122115#57122115

This has only affected two users over the course of 5 months (I have a couple of thousand users a day, but not everyone uses the maps), so really rare issue.

**Comment by mulderpf** [comment]
I suspect this might not be an issue any more.  I just saw this happen to two more users in the past week, but this is on a piece of code which was using version 0.5.0 and it was replaced back in May already.  That is what all four users who have experienced this have in common - v0.5.0 (and three of them were using Amazon devices and one was using an iPhone).

**Comment by johnpryan** [comment]
Hmm... That sounds like there could be something causing the network request to be failing. There could be more information in the logs, but it sounds like this is happening an a flutter app that has been built in release mode?

**Comment by DoryFazna** [comment]
> I copy and paste the demo (editing the token) but the map is blank, i can see the flutter logo
> the console print this "Another exception was thrown: Couldn't download or retreive file."

met with the same error just now..

**Comment by fuadarradhi** [comment]
**subdomains** will fix this error

`new TileLayerOptions(
          urlTemplate: "https://{s}.google.com/vt/lyrs=m&x={x}&y={y}&z={z}",
          subdomains:['mt0','mt1','mt2','mt3'],
)`

**Comment by ibrierley** [comment]
I've commented on another thread I opened. One thing that happened to me was that there were 2 AndroidManifest.xml files, one for debug and one main (I assume for release).

The none debug one didn't have internet enabled, so all the tiles were downloaded from cache only (debugging the cache provider failed with no dns warning).

Adding internet to the release AndroidManifest.xml file ie https://stackoverflow.com/questions/2169294/how-to-add-manifest-permission-to-an-application

I note in the example app it has internet enabled ok.
